### PR TITLE
commander add parameter COM_ARM_MIS_REQ

### PR DESCRIPTION
**Pull request by dagar**
- new parameter COM_ARM_MIS_REQ to require valid mission to arm (off by default)

### CODE REVIEW

**Comment by Antiheavy** [comment]
Flight tested this today on a custom version of Master.  Correctly blocked arming when uploading invalid missions.

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -409,10 +411,12 @@ int commander_main(int argc, char *argv[])
 
 	if (!strcmp(argv[1], "check")) {
 		int checkres = 0;
-		checkres = preflight_check(&status, &mavlink_log_pub, false, true, &status_flags, &battery, false, hrt_elapsed_time(&commander_boot_timestamp));
+		checkres = preflight_check(&status, &mavlink_log_pub, false, true, &status_flags, &battery, true, false, hrt_elapsed_time(&commander_boot_timestamp));
```
Comment (julianoes) [inline src/modules/commander/commander.cpp]: This changes the behaviour of check, on purpose?

Comment (dagar) [inline src/modules/commander/commander.cpp]: It seemed arbitrary and this is in line with the default value.

Comment (julianoes) [inline src/modules/commander/commander.cpp]: Cool, all good then.

**Pull request merged**